core/src/test/java/org/elasticsearch/common/rounding/TimeZoneRoundingTests.java

```diff
@@ -527,11 +577,15 @@ public void testDST_END_Edgecases() {
 
         // Sunday, 29 October 2000, 01:00:00 clocks were turned backward 1 hour
         // to Sunday, 29 October 2000, 00:00:00 local standard time instead
+        // which means there were two midnights that day.
 
         long midnightBeforeTransition = time("2000-10-29T00:00:00", tz);
+        long midnightOfTransition = time("2000-10-29T00:00:00-01:00");
+        assertEquals(60L * 60L * 1000L, midnightOfTransition - midnightBeforeTransition);
         long nextMidnight = time("2000-10-30T00:00:00", tz);
 
-        assertInterval(midnightBeforeTransition, nextMidnight, rounding, 25 * 60, tz);
+        assertInterval(midnightBeforeTransition, midnightOfTransition, rounding, 60, tz);
```

Comment:
Just to clarify this: this means that when we do e.g. daily date histograms in this time zone around this time change, we would get a 24h bucket for Sat 28th, then an extra 1-hour bucket for the first hour after midnight on the 29th and then a full 24h bucket for the day after the time change? Sounds like an okay trade-off to me, with the potential disadvantage that we have a very small bucket (might be confusing in stats in graphs etc, but I guess something has to five...) and that October then 32 buckets.